tagging_intents_by_intent_name

```diff
@@ -0,0 +1,117 @@
+        error_msg = "Both add_tag and delete_tag are empty. Atleast one should be provided"
+        raise  Exception(error_msg) # pylint: disable=broad-exception-raised
+
+    hf_api = humanfirst.apis.HFAPI(username=username, password=password)
+    playbook_dict = hf_api.get_playbook(namespace, playbook)
+    labelled_workspace = humanfirst.objects.HFWorkspace.from_json(playbook_dict,delimiter=delimiter)
+
+    if add_tag != "":
+        for idx,_ in labelled_workspace.tags.items():
+            if labelled_workspace.tags[idx].name == add_tag:
+                add_hf_tag =  labelled_workspace.tags[idx]
+                break
+        else:
+            add_hf_tag = labelled_workspace.tag(tag=add_tag)
+            hf_api.create_tag(namespace=namespace,
+                            playbook=playbook,
+                            tag_id=add_hf_tag.id,
+                            tag_name=add_hf_tag.name,
+                            tag_color=add_hf_tag.color)
+
+            msg1 = f"\'{add_hf_tag.name}\' is created for the playbook - \'{playbook}\'"
+            msg2 = f"in the namespace - \'{namespace}\'"
+            print(f"{msg1} {msg2}")
+    else:
+        add_hf_tag = labelled_workspace.tag(tag=add_tag)
+
+    # remove color from add_hf_tag
+    # this is beacuse the tags inside intent should have only the id and name
+    add_hf_tag = add_hf_tag.to_dict()
+    del add_hf_tag["color"]
+
+    for intent_name,_ in labelled_workspace.intents.items():
+        add_flag = 0
+        del_flag = 0
+        if intent_name.find(search_string) != -1:
+            # check if the tag is already present in the intent
+            intent_tags = labelled_workspace.intents[intent_name].tags
+            intent_tags_after_del = []
+            for _, intent_tag in enumerate(intent_tags):
+                intent_tag = intent_tag.to_dict()
+
+                # color peroperty should be removed from the tags inside intent property
+                # TODO: update humanfirst objects to not set the color property in the tags inside intent property
+                del intent_tag["color"]
+
+                if add_hf_tag["name"] != "":
+                    if intent_tag["name"] == add_hf_tag["name"]:
+                        print(f"\'{add_hf_tag['name']}\' tag is already present in the intent \'{intent_name}\'")
+                        add_flag = 1
+                else:
+                    add_flag = 1
+
+                if intent_tag["name"] == delete_tag:
+                    del_flag = 1
+                    continue
+
+                intent_tags_after_del.append(intent_tag)
+
+            labelled_workspace.intents[intent_name].tags = intent_tags_after_del
+
+            if del_flag == 0 and delete_tag != "":
+                error_msg = f"{delete_tag} is not present in any of the intents"
+                raise Exception(error_msg) # pylint: disable=broad-exception-raised
+
+            if add_flag == 0 or del_flag == 1:
```

Comment:
recommend elif, then an elif on del_flag = 0 and add_flag = 0 which is the nothing to do correct step, then catch all else which throws an error if it isn't caught by the above ones catches any logical errors and throws and exception.